packages/web-app-files/tests/unit/components/SideBar/Shares/Collaborators/EditCollaborator.spec.js

```diff
@@ -0,0 +1,336 @@
+    })
+
+    it('should render the save button', () => {
+      const wrapper = getShallowMountedWrapper({ user: 'user0' })
+      const saveButton = wrapper.find(selectors.saveButton)
+      expect(saveButton.exists()).toBe(true)
+      expect(saveButton.attributes().disabled).toBe('true')
+      expect(saveButton.text()).toBe('Save')
+    })
+
+    it('cancel button should work', async () => {
+      const cancelTrigger = jest.spyOn(Editcollaborator.methods, '$_ocCollaborators_cancelChanges')
+
+      const wrapper = getMountedWrapper({ user: 'user0' })
+      const cancelButton = wrapper.find(selectors.cancelButton)
+
+      expect(cancelTrigger).not.toHaveBeenCalled()
+      await cancelButton.trigger('click')
+      expect(cancelTrigger).toHaveBeenCalledTimes(1)
+    })
+
+    it('save button should work', async () => {
+      const saveTrigger = jest.spyOn(Editcollaborator.methods, '$_ocCollaborators_saveChanges')
+
+      const wrapper = getMountedWrapper({ user: 'user0' })
+      const saveButton = wrapper.find(selectors.saveButton)
+
+      const roleSelect = wrapper.findComponent(VueSelect)
+      expect(roleSelect.props('value')).toMatchObject(roles.viewer)
+      expect(wrapper.vm.selectedRole).toMatchObject(roles.viewer)
+
+      await roleSelect.vm.select(roles.editor)
+      expect(roleSelect.props('value')).toMatchObject(roles.editor)
+      expect(wrapper.vm.selectedRole).toMatchObject(roles.editor)
+
+      expect(saveTrigger).not.toHaveBeenCalled()
+      await saveButton.trigger('click')
+      expect(saveTrigger).toHaveBeenCalledTimes(1)
+    })
+
+    it('should not display saving in progress indicator by default', () => {
+      const wrapper = getMountedWrapper({ user: 'user0' })
+
+      const spinner = wrapper.find(selectors.savingInProgress)
+      expect(spinner.exists()).toBe(false)
+    })
+
+    it('should display the loading indicator when saving is in progress', async () => {
+      const wrapper = getMountedWrapper({ user: 'user0' })
+
+      await wrapper.setData({ saving: true })
+      const spinner = wrapper.find(selectors.savingInProgress)
+      expect(spinner.exists()).toBe(true)
+    })
+  })
+})
+
+function getResource({ filename = 'testFile', extension = 'txt', type = 'file', owner = 'user0' }) {
+  return {
+    id: '4',
+    fileId: '4',
+    icon: type,
+    name: type === 'file' ? `${filename}.${extension}` : filename,
+    extension: extension,
+    path: type === 'file' ? `/${filename}.${extension}` : `/${filename}`,
+    type,
+    mdate: 'Mon, 12 Jul 2021 11:04:33 GMT',
+    size: '163',
+    indicators: [],
+    permissions: 'RDNVW',
+    starred: false,
+    etag: '"89128c0e8122002db57bd19c9ec33004"',
+    shareTypes: [],
+    downloadURL: '',
+    ownerDisplayName: displayNames[owner],
+    ownerId: owner,
+    canDownload: () => true,
+    isReceivedShare: () => true,
+    canBeDeleted: () => true,
+    canRename: () => true,
+    canShare: () => true
```

Comment:
imo `type` is the only required attribute here, can you review this and use only required